### PR TITLE
Miguel/fixes prerelease

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

Fixes suggested by @loomlike  and documentation added to deeprec algos

### Checklist:

- [ ] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: reco_utils/recommender/deeprec/models/base_model.py [hunk 1]
```diff
@@ -46,24 +50,33 @@ def __init__(self, hparams, iterator_creator, graph=None):
...
-        """
-        Make final output as prediction score, according to different tasks.
-        :param logit: base prediction value
-        :param task:  regression or classification
+        """Make final output as prediction score, according to different tasks.
+        
+        Args:
+            logit (obj): Base prediction value.
+            task (str): A task (values: regression/classification)
         :return: transformed score
```
Comment (loomlike) [inline reco_utils/recommender/deeprec/models/base_model.py]: Here, you missed one line of reformatting. `Returns:`

File: reco_utils/recommender/deeprec/models/base_model.py [hunk 2]
```diff
@@ -385,16 +358,16 @@ def load_model(self, model_path=None):
             raise IOError("Failed to find any matching files for {0}".format(act_path))
 
     def fit(self, train_file, valid_file, test_file=None):
-        """
-        Fit the model with train_file. Evaluate the model on valid_file per epoch to observe the training status.
+        """Fit the model with train_file. Evaluate the model on valid_file per epoch to observe the training status.
         If test_file is not None, evaluate it too.
+
```
Comment (loomlike) [inline reco_utils/recommender/deeprec/models/base_model.py]: This function accepting both validation and test data is strange to me... why does this function evaluate the model both on validation and test data?

Also, having validation data as an input to `fit` function doesn't sound like following single responsibility principle.
E.g. TF has a dedicated function for this ([train_and_evaluate](https://www.tensorflow.org/api_docs/python/tf/estimator/train_and_evaluate))

Comment (miguelgfierro) [reply reco_utils/recommender/deeprec/models/base_model.py]: yeah, good point, I'm discussing this with Jianxun https://github.com/Microsoft/Recommenders/issues/539 He and I will work on this in the next release

File: reco_utils/recommender/deeprec/models/base_model.py (hunk 1 above)
Comment (miguelgfierro) [reply reco_utils/recommender/deeprec/models/base_model.py]: good catch

File: tests/unit/test_python_splitter.py [hunk 3]
```diff
@@ -47,8 +47,7 @@ def python_dataset(test_specs):
     def random_date_generator(start_date, range_in_days):
         """Helper function to generate random timestamps.
```
Comment (loomlike) [inline tests/unit/test_python_splitter.py]: This file's `tech_specs["user_ids"]` is not used from anywhere.

File: tests/unit/test_python_splitter.py [hunk 4]
```diff
@@ -94,28 +86,34 @@ def python_int_dataset(test_specs):
...
     assert len(splits[2]) == round(shape * 0.52)
 
     with pytest.raises(ValueError):
-        splits = split_pandas_data_with_ratios(pandas_dummy_timestamp, ratios=[0.6, 0.2, 0.4])
+        splits = split_pandas_data_with_ratios(
+            pandas_dummy_timestamp, ratios=[0.6, 0.2, 0.4]
+        )
 
 
 def test_min_rating_filter(python_dataset):
```
Comment (loomlike) [inline tests/unit/test_python_splitter.py]: This function doesn't test min_rating_filter thoroughly as it relies on random-generated data of 1000 records for 4 users, end up almost no chances to have the users whose `min_rating<5`. Should provide a dataset which actually includes those users.

Comment (miguelgfierro) [reply tests/unit/test_python_splitter.py]: @loomlike, do you have time today to take a look at this? if not, @yueguoguo, do you have time tomorrow during your day? if not, I'll take a look tomorrow in my morning :-)

Comment (loomlike) [reply tests/unit/test_python_splitter.py]: @miguelgfierro where is plan C? haha I can take a look at this today. Will push the changes to this branch.

**Comment by miguelgfierro** [comment]
looks great!

**Pull request merged**